### ISSUE TITLE
Spacy2 Matcher receiving position of match entity from text

### ISSUE THREAD

**Issue by vladimircape** [explosion/spaCy#2544]
I have such code
```
nlp            = spacy.load('en_core_web_sm')
matcher    =  Matcher(nlp.vocab)
doc            = nlp(text)
matches    = matcher(doc)
```

I want to know at what position of text macher found match?

because when i made such code

```
            for element in matches:
                match_id, start, end = element
                print(doc[start: end])
```

I just so index of tokents from doc, not start and end index in string text.
I need it to understand from what part match is
example of sentance
"New product Galaxy S7 appear, and price of Galaxy S7 was high"
Matcher for Galaxy S7
but how i can find start and end position of this 2 same matching?

**Comment by ines** [comment]
Yes, the `start` and `end` values are the token indices. But they let you create a `Span` object (a slice of the doc) that exposes the start and end character offset as the `Span.start_char` and `Span.end_char` attributes. For example:

```python
for match_id, start, end in matches:
    span = doc[start: end]
    print(span.text, span.start_char, span.end_char)
```

**Comment by vladimircape** [comment]
After i wrote question, i realized that probably I can use match_id and put it like an index to like
`doc.ents[match_id]`
?

**Comment by ines** [comment]
I don't think that's going to work – the `match_id` is the integer ID of the matched rule, for example:

```python
matcher.add('YOUR_RULE', None, *patterns)
#              ^ integer ID of this will be the match_id
```

The matches are also separate from the `doc.ents`. But you can create a `Span` from a match and *add it* to the `doc.ents`:

```python
for match_id, start, end in matches:
    span = Span(doc, start, end, label=doc.vocab.strings['PRODUCT'])
    doc.ents = list(doc.ents) + [span]
```
